### ISSUE TITLE
Message Dates in Telegram

### ISSUE THREAD

**Issue by sattagon54** [peter-iakovlev/Telegram#206]
Hi,

I was wondering whether Telegram could implement a moving (dynamic?) date instead of the current static (stationary?) one they currently have. It can be a bit inconvenient to find the bubble where the date is written when going through Telegram chats. Especially long chats.

At the moment (V4.7 on iOS, unsure about android...) when looking through previous messages you can't see the date they were posted unless you keep scrolling and scrolling to find the little bubble with the date.

I suggest something similar to WhatsApp, in which the date appears as a bubble on the top of the screen as you scroll, and changes as you go through the messages. This would be extremely helpful for finding messages on a particular date.

I'm sorry if it sounds a bit confusing, but I don't really know how else to explain it in more simpler terms. I would be grateful if someone else could help.

An easy way to understand what I'm talking about is if you scroll through a WhatsApp conversation and see the changing date at the top, while on Telegram the date is "part of" the chat and doesnt move.

I know it's a bit nit-picky, but it's a small feature that I believe would really improve the already amazing Telegram experience.

Thanks

**Comment by HairLossTalk** [comment]
Its not nit-picky at all.  Its an incredibly important feature and I'm blown away that Telegram has gone this long with literally no date stamps on the messages.  

In other apps you can slide the messages to the right and it reveals the date.  Others you can click and hold on the message bubble.  But to completely leave dates out aside from one bubble at the beginning of the 24 hour period, is ridiculous.   

I am in many Crypto ICO chats and there are literally thousands of messages each 24 hour period.   If there is a Pinned message you simply have no way of knowing what day it was posted.  

I came here today googling about this because for months it has frustrated me that there are no dates on anything, unless you do a search.  Then weirdly, there's a date stamp that shows on the left side in the Desktop app.

Please fix this, Telegram.

**Comment by thedailymom** [comment]
I agree! It's really a little (maybe more than a little) inconvenient tbh. I hope telegram fix this soon.

**Comment by Younes-L** [comment]
This has been addressed a while ago
